src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -32,193 +32,614 @@
+                this.dataImportFieldMappingSetRecordName = null;
+            }
+
+            // Convert an empty string ('') value into null for the entity reference to save properly.
+            if (String.isEmpty(this.targetObjectMappingRecordName)) {
+                this.targetObjectMappingRecordName = null;
+            }
+
+            Map<String, Object> fieldValues;
+            if (this.isDeleted != true) {
+                fieldValues = new Map<String, Object>{
+                    'Data_Import_Field_Mapping_Set__c' => dataImportFieldMappingSetRecordName == null ? 'Default_Field_Mapping_Set' : dataImportFieldMappingSetRecordName,
+                    'Required__c' => isRequired,
+                    'Source_Field_API_Name__c' => sourceFieldAPIName,
+                    'Target_Field_API_Name__c' => targetFieldAPIName,
+                    'Target_Object_Mapping__c' => targetObjectMappingRecordName,
+                    'Is_Deleted__c' => isDeleted
+                };
+            } else {
+                fieldValues = new Map<String, Object>{ 'Is_Deleted__c' => isDeleted };
+            }
+            customMd = CMT_MetadataAPI.addFieldsToMetadataObject(customMd, fieldValues);
+
+            return customMd;
+        }
+
+        /*******************************************************************************************************
+        * @description Utility to convert a name into a valid 'DeveloperName' quality value (or unique GUID)
+        * for the CMT record
+        *
+        * @param name String
+        * @param makeUnique Boolean
+        * @return DeveloperName format string
+        */
+        private String generateRecordName(String name, Boolean makeUnique, String objectLabel, String objectAPIName) {
+            name = name.replaceAll('[^\\w]+', '_').replaceAll('_{2,}', '_');
+
+            if (String.isNotBlank(objectLabel)) {
+                objectLabel = objectLabel.replace(' ', '_');
+                name = name.replace(objectAPIName, objectLabel);
+            }
+
+            if (name.length() > 30) {
+                String firstFifteen = name.substring(0, 15);
+                String lastFifteen = name.substring(name.length() - 15, name.length());
+                name = firstFifteen + lastFifteen;
+            }
+
+            if (makeUnique == true) {
+                // Create a GUID format unique string
+                Blob b = Crypto.GenerateAESKey(128);
+                String h = EncodingUtil.ConvertTohex(b);
+                // force everything to have a 9-char random string at the end
+                name = name.left(30).removeEnd('_') + '_' + h.left(9);
+            } else {
+                name = name.left(40);
+            }
+
+            return name;
+        }
     }
 
+    /*******************************************************************************************************
+    * @description Wrapper class for help text field
```

Comment:
How about "Wrapper class for field mapping information stored in the inline help text of a field"?